html/includes/authenticate.inc.php

```diff
@@ -96,7 +96,8 @@
 
         $permissions = permissions_cache($_SESSION['user_id']);
         if (isset($_POST['username'])) {
-            header('Location: '.$_SERVER['REQUEST_URI'] ?: $config['base_url'], true, 303);
+            // Trim the trailing slash off of base_url and concatenate the (relative) REQUEST_URI
+            header('Location: '.substr($config['base_url'], 0, -1).$_SERVER['REQUEST_URI'], true, 303);
```

Comment:
This variable does not always have a slash at the end.
Please add a sanity check before cutting off a character from the base_url; worst case you kill the tld.